tests/func/test_scm.py

```diff
@@ -61,112 +62,93 @@ def test_commit_in_submodule(self):
         self.assertTrue("foo" in self.git.git.ls_files())
 
 
-class TestIgnore(object):
-    @staticmethod
-    def _count_gitignore_entries(string):
-        lines = get_gitignore_content()
-        return len([i for i in lines if i == string])
+def _count_gitignore_entries(line):
+    lines = get_gitignore_content()
+    return lines.count(line)
 
-    def test_ignore(self, git, repo_dir):
-        git = Git(repo_dir._root_dir)
-        foo = os.path.join(repo_dir._root_dir, repo_dir.FOO)
 
-        target = "/" + repo_dir.FOO
+def test_ignore(tmp_dir, scm):
+    tmp_dir.gen({"foo": "foo contents"})
 
-        git.ignore(foo)
-        assert os.path.isfile(Git.GITIGNORE)
-        assert self._count_gitignore_entries(target) == 1
+    foo = fspath(tmp_dir / "foo")
+    target = "/foo"
 
-        git.ignore(foo)
-        assert os.path.isfile(Git.GITIGNORE)
-        assert self._count_gitignore_entries(target) == 1
+    scm.ignore(foo)
+    assert os.path.isfile(Git.GITIGNORE)
```

Comment:
in latter tests, we use ".gitignore", lets use it or change the rest into `Git.GITIGNORE`

Reply:
we can also replace `os.path.isifle` -> (tmp_dir / ".gitignore").is_file(), but it does not seem it will change much.